src/cargo/ops/cargo_compile.rs

```diff
@@ -475,8 +475,13 @@ pub fn create_bcx<'a, 'cfg>(
                 extra_args = Some(args);
             }
 
-            if let Some(args) = extra_args {
-                extra_compiler_args.insert(unit.clone(), args.clone());
+            if let Some(mut args) = extra_args {
+                match extra_compiler_args.get_mut(&unit) {
+                    None => {
+                        extra_compiler_args.insert(unit.clone(), args);
+                    }
+                    Some(existing) => existing.append(&mut args),
+                }
```

Comment:
I feel like this could be slightly easier to read with the `entry` API.  Maybe something like this:

```rust
extra_compiler_args
    .entry(unit.clone())
    .or_default()
    .extend(args);
```

(Just a minor stylistic preference.)

Reply:
Using the entry API requires cloning unit even if there were existing args. I can change it if you like though.

Reply:
Yea, it is an Rc clone, so it is very cheap.

Reply:
I changed this to use entry. You're right, the code looks a lot nicer now :)